Tests/Common/Data/SliceTests.cs

```diff
@@ -506,5 +507,580 @@ public void AccessesTradeBarAndQuoteBarForSameSymbol()
+
+                var result = -1;
+                Assert.DoesNotThrow(() => result = test(GetSlice(), Symbols.SPY));
+                Assert.AreEqual(2, result);
+
+                result = -1;
+                Assert.DoesNotThrow(() => result = test(GetPythonSlice(), Symbols.SPY));
+                Assert.AreEqual(2, result);
+            }
+        }
+
+        [Test]
+        public void PythonSlice_copy()
+        {
+            using (Py.GIL())
+            {
+                dynamic test = PythonEngine.ModuleFromString("testModule",
+                    @"
+from clr import AddReference
+AddReference(""QuantConnect.Common"")
+from QuantConnect import *
+
+def Test(slice, symbol):
+    copy = slice.copy()
+    return ', '.join([f'{k}: {v.Value}' for k,v in copy.items()])").GetAttr("Test");
+
+                var result = string.Empty;
+                Assert.DoesNotThrow(() => result = test(GetPythonSlice(), Symbols.SPY));
+                Assert.AreEqual("SPY R735QTJ8XC9X: 10.0, AAPL R735QTJ8XC9X: 11.0", result);
+            }
+        }
+
+        [Test]
+        public void PythonSlice_items()
+        {
+            using (Py.GIL())
+            {
+                dynamic test = PythonEngine.ModuleFromString("testModule",
+                    @"
+from clr import AddReference
+AddReference(""QuantConnect.Common"")
+from QuantConnect import *
+
+def Test(slice):
+    return ', '.join([f'{k}: {v.Value}' for k,v in slice.items()])").GetAttr("Test");
+
+                var result = string.Empty;
+                Assert.DoesNotThrow(() => result = test(GetPythonSlice()));
+                Assert.AreEqual("SPY R735QTJ8XC9X: 10.0, AAPL R735QTJ8XC9X: 11.0", result);
+            }
+        }
+
+
+        [Test]
+        public void PythonSlice_keys()
+        {
+            using (Py.GIL())
+            {
+                dynamic test = PythonEngine.ModuleFromString("testModule",
+                    @"
+from clr import AddReference
+AddReference(""QuantConnect.Common"")
+from QuantConnect import *
+
+def Test(slice):
+    return slice.keys()").GetAttr("Test");
+
+                var slice = GetPythonSlice();
+                var result = new List<Symbol>();
+                Assert.DoesNotThrow(() => result = test(slice));
+                foreach (var key in slice.Keys)
+                {
+                    Assert.IsTrue(result.Contains(key));
+                }
+            }
+        }
+
+        [Test]
+        public void PythonSlice_values()
+        {
+            using (Py.GIL())
+            {
+                dynamic test = PythonEngine.ModuleFromString("testModule",
+                    @"
+from clr import AddReference
+AddReference(""QuantConnect.Common"")
+from QuantConnect import *
+
+def Test(slice):
+    return slice.values()").GetAttr("Test");
+
+                var slice = GetPythonSlice();
+                var result = new List<BaseData>();
+                Assert.DoesNotThrow(() => result = test(slice));
```

Comment:
Actually, no. It is failing because we got:
```csharp
protected override IEnumerable<dynamic> GetValues
    => GetKeyValuePairEnumerable().Select(data => (dynamic)data);
```
instead of
```csharp
protected override IEnumerable<dynamic> GetValues
    => GetKeyValuePairEnumerable().Select(data => (dynamic)data.Value);
```